### ISSUE TITLE
Use of FP16 in backward() with create_graph = True?

### ISSUE THREAD

**Issue by sjscotti** [pytorch/pytorch#64655]
Hi
I am using pytorch within a chatbot training routine and I would like to get FP16’s advantages in GPU memory/speed.   My routine seems to work fine using FP32.  In switching it to FP16, my problem appears to be caused by the `loss.backward()`  pytorch routine.  For my optimizer to work, I need to use the argument `create_graph = True` in  `backward`.  When I do that with the model I am working with (for any scale factor on the loss that I’ve tried), `backward` gives `nan` for some of the parameter gradients while others appear to have reasonable values.  However, when I use the argument `create_graph = False` to see how these gradients compare with the former case, there doesn't appear to be any `nan` generated and the gradients that weren't `nan` in the former case agree with what is computed here.  Since the `create_graph` is supposed to allow computation of higher order derivatives (but not the first derivative), I am confused why the `create_graph = True` argument is having this effect.  Any suggestions you can provide would be appreciated!

cc @ngimel

**Comment by VitalyFedyunin** [comment]
Hello!

It is better to ask such kinds of questions at https://discuss.pytorch.org/
I am going to close this issue. Feel free to reopen it, if you think that you need to report a bug or make a feature proposal.

Thanks.

**Comment by sjscotti** [comment]
Thanks 
Issue created in discuss.pytorch.org

see...
https://discuss.pytorch.org/t/use-of-fp16-in-backward-causes-nan-in-gradients-only-when-create-graph-true/131473?u=sjscotti
